haddock-api/src/Haddock/GhcUtils.hs

```diff
@@ -112,6 +118,14 @@ sigNameNoLoc (InlineSig    n _)        = [unLoc n]
 sigNameNoLoc (FixSig (FixitySig ns _)) = map unLoc ns
 sigNameNoLoc _                         = []
 
+
+uniquifyName :: SetName name => name -> name
+uniquifyName = liftA2 setName (updateName . getName) id
+  where
+    updateName = liftA2 setNameUnique id $
+        liftA2 deriveUnique id ((+1) . getKey) . nameUnique
```

Comment:
This is a bit suspicious. That said, I think the smell is coming more from the fact that `deriveUnique` exists... and I don't see another easy workaround.